### PR TITLE
add setuptools_scm to requirements

### PR DESCRIPTION
**Pull request by mkolopanis**

### CODE REVIEW

File: .github/workflows/publish_to_pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: Setup Mambaforge
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+            miniforge-variant: Mambaforge
+            miniforge-version: latest
+            use-mamba: true
+            python-version: ${{ env.PYTHON }}
+            environment-file: ci/${{ env.ENV_NAME }}.yaml
+            activate-environment: ${{ env.ENV_NAME }}
```
Comment (steven-murray) [inline .github/workflows/publish_to_pypi.yaml:32]: can't you just use simple `setup-python` action here? Why do you need conda/mamba?

File: simpleDS/branch_scheme.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    If and when this is added to setuptools_scm this function and file can be removed.
+    """
+    if version.exact or version.node is None:
+        return version.format_choice("", "+d{time:{time_format}}", time_format="%Y%m%d")
+    else:
+        if version.branch == "main":
+            return version.format_choice("+{node}", "+{node}.dirty")
+        else:
+            return version.format_choice("+{node}.{branch}", "+{node}.{branch}.dirty")
```
Comment (steven-murray) [inline simpleDS/branch_scheme.py:20]: do you really *need* the branch name for absolute reproducibility, or does the hash uniquely define everything you need?

Comment (mkolopanis) [reply simpleDS/branch_scheme.py:20]: I think the use of branch scheme is mostly lazy copy-pasta. We could probably get away with just the hash on this repo.

Comment (mkolopanis) [reply simpleDS/branch_scheme.py]: I've deleted the branch scheme

Comment (steven-murray) [reply simpleDS/branch_scheme.py]: 👍🏻

**Review by steven-murray — APPROVED** [review]

**Comment by mkolopanis** [comment]
Checks aren't reporting at all but if you click on the commit i can see them. Not really sure what's going on with that.

**Pull request merged**